### DISCUSSION TITLE
[New Feature] Implement viewport zooming

### DISCUSSION THREAD

**Discussion by Rampastring** [Vinifera-Developers/Vinifera#1494] [Feature Requests]
### Description:
The game has a feature that allows you to zoom in on the map. It could be handy for giving the player more precision in micromanagement when they need it. However it was left unfinished by Westwood.

I can't remember how exactly the zoom feature is broken, maybe someone else could fill in the details?

### Additional Files:
TS-patches has a hack that can enable zooming through a hotkey:
https://github.com/CnCNet/ts-patches/blob/master/src/tactical_zoom.c

Having a hotkey for zooming would be beneficial if the zoom feature can be made functional.

**Comment by CCHyper** [comment]
The zoom feature has two issues;

1. The mouse is drawn before and after the blit process, which causes the unscaled/zoomed mouse to flicker.
2. The position of the mouse is calculated incorrectly relative to the zoom level.

I would say this is a low priority feature, as it could take up too much research time for a very little payoff...

But I agree, this is a nice feature to have.
